### PR TITLE
Adds support for custom namespace delimiters

### PR DESCRIPTION
**Pull request by raventid**
I've decided to get back to namespace delimiters topic. My solution is extremely straightforward, but one small detail makes me nervous. Why we've been splitting string and taking last element? We always need just one last identifier, so it would be simpler to just cut it out of string. I did this.

One thing I don't like about this solution is that it gives you too much freedom (to write 643826%$&^/my_dependency). But dealing with this is not the responsibility of dependency mapper, so leave it now.

### CODE REVIEW

**Comment by raventid** [comment]
One interesting advantage of custom separators is that it's easier to navigate in RubyMine. You can quickly copy string from Import and jump to definition, lol

File: lib/dry/auto_inject/dependency_map.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module Dry
   module AutoInject
     DuplicateDependencyError = Class.new(StandardError)
+    DependencyNameIsNotValid = Class.new(StandardError)
```
Comment (timriley) [inline lib/dry/auto_inject/dependency_map.rb]: Perhaps we could shorten this to `DependencyNameInvalid`?

**Comment by solnic** [comment]
> You can quickly copy string from Import and jump to definition

FWIW, it should be fairly easy to write plugins for editors/ides so that you can quickly navigate to a file by either clicking on identifier or using a key shortcut/editor command.

**Comment by raventid** [comment]
@timriley done.

**Comment by raventid** [comment]
@solnic it should be, but it's not so straightforward. Besides, it is a typical task, you think, someone else will do better and sooner :)

**Comment by timriley** [comment]
Thanks @raventid!

**Comment by timriley** [comment]
And released in 0.4.4
